lib/puppet/provider/group/groupadd.rb

```diff
@@ -30,23 +30,21 @@ def gid
 
   def findgroup(key, value)
     group_file = "/etc/group"
-    group_keys = ['group_name', 'password', 'gid', 'user_list']
+    group_keys = [:group_name, :password, :gid, :user_list]
```

Comment:
This actually might be a behavior change that might break things for users. I have no idea if anyone calls  `findgroup` in a manifest or something in custom puppet code, but it's possible.

@binford2k do you know if this is something we can check?

```
irb(main):001:0> group_keys = [:group_name, :password, :gid, :user_list]
=> [:group_name, :password, :gid, :user_list]
irb(main):002:0> group_keys.index(:gid)
=> 2
irb(main):003:0> group_keys.index('gid')
=> nil
```

Reply:
I agree it's nicer to have this as an array of symbols, but I think it's safer to keep it as an array of strings

Reply:
It's definitely a grey area. The `findgroup` method is not intended to be public API, but the method isn't marked private, so you never know. Keeping it as strings is probably safer. Might also want to mark methods as private (at least in yardoc using `@private`).

Reply:
@melissa what I can say is that no Puppet manifests in public GitHub module repositories contain the literal string `findgroup`. I don't at the moment have a way to introspect into Ruby files in the modules, but if that were useful I could do it in ~ 30m or so.

Reply:
@melissa I did the same for the useradd provider in a pull requests from a couple of months ago (https://github.com/puppetlabs/puppet/pull/7768), so it would be nice to have the same behavior here. I understand that this method (as is) is public, but it was never documented as such. I searched this method across ruby files on GitHub, and apart from Puppet forks I couldn't find anything relevant, and it hasn't been documented by YARD.

I agree that we should not break existing users but I believe this obscure method is not used of Puppet... at least not publicly. I would go forward with this change and also mark the method as private in Ruby.

Reply:
Okay, I'm glad this has been thought through. If you're confident in merging this in, then I agree.

Reply:
I marked the method as private and adjusted the spec tests accordingly, I'm good to merge :)